templates/cicd/pipeline.yml

```diff
@@ -6,18 +6,19 @@ version: {{.Version}}
 # This section defines the source artifacts.
 source:
   # The name of the provider that is used to store the source artifacts.
-  provider: {{.Provider}}
+  provider: {{.Source.ProviderName}}
   # Additional properties that further specifies the exact location
   # the artifacts should be sourced from.
   properties:
```

Comment:
Can we get rid of this properties struct, and have the repo and branch be at the same level? (can we also add branch plz :D)

Reply:
Added the `branch` field.

I don't think we will be able to remove the `properties` field and replace it with explicit fields because the necessary set of fields for each provider is different. Using a `map[string]interface{}` in Go is the best way I can think of to handle dynamic "map" in YAML.

Reply:
Oky doky. We don't have to worry about it too much. Keep it in the back of your head - if we can get think of a way to get rid of it.